library/HTMLPurifier/AttrDef/URI/Host.php

```diff
@@ -80,7 +80,7 @@ public function validate($string, $config, $context)
         // as per RFC 3696, the top label need only not be all numeric.
         // The productions describing this are:
         $a   = '[a-z]';     // alpha
-        $an  = '[a-z0-9]';  // alphanum
+        $an  = "[a-z0-9$underscore]";  // alphanum
```

Comment:
can you update the comment above too?

Reply:
Would it be ok to update the comment as follows?
```
        $an  = "[a-z0-9$underscore]";  // alphanum | "_"
        $and = "[a-z0-9-$underscore]"; // alphanum | "-" | "_"
```

Reply:
oops, I meant

```
        // This doesn't match I18N domain names, but we don't have proper IRI support,
        // so force users to insert Punycode.
        // There is not a good sense in which underscores should be
        // allowed, since it's technically not! (And if you go as
        // far to allow everything as specified by the DNS spec...
        // well, that's literally everything, modulo some space limits
        // for the components and the overall name (which, by the way,
        // we are NOT checking!).  So we (arbitrarily) decide this:
        // let's allow underscores wherever we would have allowed
        // hyphens, if they are enabled.  This is a pretty good match
        // for browser behavior, for example, a large number of browsers
        // cannot handle foo_.example.com, but foo_bar.example.com is
        // fairly well supported.
```

Reply:
Thank you for your confirmation and merging!